logdetective/logdetective.py

```diff
@@ -64,11 +64,15 @@ def main():
     log = retrieve_log_content(args.file)
     log_summary = extractor(log)
 
-    ratio = len(log_summary.split('\n')) / len(log.split('\n'))
-    LOG.debug("Log summary: \n %s", log_summary)
+    ratio = len(log_summary) / len(log.split('\n'))
+
     LOG.info("Compression ratio: %s", ratio)
 
     LOG.info("Analyzing the text")
+
+    log_summary = format_snippets(log_summary)
+    LOG.debug("Log summary: \n %s", log_summary)
```

Comment:
please change to `.info` because no snippets are printed by default (without `-v`) and it just looks awkward:
```
$ logdetective cockpit.log
INFO:logdetective:Getting summary
INFO:logdetective:Compression ratio: 0.008032128514056224
INFO:logdetective:Analyzing the text
Explanation:
[Snippet No. 0, Snippet No. 1, Snippet No. 2, Snippet No. 3, Snippet No. 4, Snippet No. 5]: These snippets represent successful copying or moving of log files from one lo
cation to another. No error or failure occurred during these operations.

[Snippet No. 6]: This snippet indicates that a script or process executed with exit code 2. While an exit code of 0 indicates successful execution, an exit code of 2 typically signifies a "run-time error" or "system error". However, without more context about the script or process that generated this exit code, it's impossible to pinpoint
 the exact cause of the failure.

[Snippet No. 7]: This snippet indicates that a connection to the IP address 18.191.227.80 has been closed. It doesn't necessarily mean there was an error or failure during the build process; it could simply mean that the connection was no longer needed or was terminated for some reason. However, it's possible that this connection was important for the build process, so it's worth investigating further to determine if its closure had an impact on the build outcome.

Complete explanation:

Based on the provided log snippets, there are no clear indications of build failures or errors. However, there are some hints that there might have been issues with connections or dependencies during the build process:

1. Snippet No. 6 indicates that a script or process exited with an error code (exit code 2). Without more context about the script or process, it's impossible to determine what went wrong. It could be an issue with the script itself, or it could be related to dependencies or environment variables that were not properly set up.
2. Snippet No. 7 suggests that a connection to an IP address was closed during the build process. Depending on the nature of the connection and what was being transferred
 over it, this could potentially impact the build outcome. For example, if the connection was required to download necessary dependencies or to communicate with remote servers or services, its closure could cause build failures or errors

...
```

Reply:
Good point.